spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationJob.scala

```diff
@@ -473,5 +498,34 @@ object StandardizationJob {
     )
   }
 
+  /**
+    * Ensures that the 'spark.sql.columnNameOfCorruptRecord' Spark setting is set to field name not present in the
+    * output schema
+    * @param desiredSchema the desired output schema
+    * @param cmd           program settings
+    * @param spark         current Spark session
+    * @return              if columnNameOfCorruptRecord is to be used then the name of the column, if not and rather
+    *                      fail fast, function returns None
+    */
+  private def ensureUniqueCorruptRecordFieldName(desiredSchema: StructType, cmd: StdCmdConfig)
+                                                (implicit spark: SparkSession): Option[String] = {
+    val defaultResult = spark.conf.get(ColumnNameOfCorruptRecordConf)
+    var result = defaultResult
+    var i = 1
+    while (SchemaUtils.getField(result, desiredSchema).nonEmpty) {
+      result = defaultResult + i.toString
+      i += 1
+    }
+    if (defaultResult != result) {
+      spark.conf.set(ColumnNameOfCorruptRecordConf, result)
+    }
+    if (cmd.failOnInputNotPerSchema) {
```

Comment:
Still want to avoid conflict of the special column name with an existing column in the schema.

Reply:
All of the following is debatable, it's up to you:
* Maybe change the function to produce a non-optional result.
* Since `cmd` is used only for this check it looks like this check should be done outside this function. And you won't need to pass `cmd` which will make it more reusable.
* `desiredSchema` is an odd name of the parameter here, maybe just `schema`.
* This function might belong in `Utils` since it serves a very general-purpose task.

Reply:
Good point making the function widely available. Probably should be used in **Conformance** too just to be sure.